pycalphad/tests/test_energy.py

```diff
@@ -33,22 +40,35 @@ def calculate_energy(model, variables, mode='sympy'):
     variables, dict
         Dictionary of all input variables.
 
+    output : str
+        String of the property to calculate, e.g. 'ast'
+
     mode, ['numpy', 'sympy'], optional
         Optimization method for the abstract syntax tree.
     """
-    # Generate a callable energy function
+    # Generate a callable function
     # Normally we would use model.subs(variables) here, but we want to ensure
     # our optimization functions are working.
-    energy = make_callable(model.ast, list(variables.keys()), mode=mode)
+    prop = make_callable(getattr(model, output), list(variables.keys()), mode=mode)
```

Comment:
`variables` is not guaranteed to have stable ordering for keys and values on all supported platforms (at least until py27 rolls off). It should probably be wrapped to `OrderedDict` at this time.

Reply:
According to this [SO answer](https://stackoverflow.com/a/835430):

>> If `items()`, `keys()`, `values()`,  `iteritems()`, `iterkeys()`, and  `itervalues()` are called with no intervening modifications to the dictionary, the lists will directly correspond.

> On [2.x documentation](https://docs.python.org/2/library/stdtypes.html#dict.items) and [3.x documentation](https://docs.python.org/3/library/stdtypes.html#dictionary-view-objects)

So I think in this case, the lines

```python
    prop = make_callable(getattr(model, output), list(variables.keys()), mode=mode)
    # Unpack all the values in the dict and use them to call the function
    return prop(*(list(variables.values())))
```

Should preserve a consistent ordering, even though they are not guaranteed to be in insertion order, as OrderedDict gives. If this didn't work `zip(d.keys(), d.values())` would be unstable.